### DISCUSSION TITLE
Pregunta 1 parte 1 Duda ejemplo capsula

### DISCUSSION THREAD

**Discussion by Juanfischermontero** [IIC2133-PUC/2024-2#150] [Tarea 2] [answered]
En la capsula, para la pregunta de rango de peso, se sugiere crear un arbol de indices a partir de la lista ya ordenada. Se muestra este ejemplo:
![image](https://github.com/user-attachments/assets/5a486d6d-30ea-40bd-992b-70b54e1bf20c)
Entiendo la construccion pero no entiendo la busqueda. Si trato de buscar el 3 como un arbol binario normal: parto por arriba, 3 es mayor a 2, me voy para la derecha. 3 es menor que 4, me voy para la izquierda. No llegue al 3. No estoy entendiendo algo?

**Comment by CarlangaUC** [comment, marked as answer]
Hola! 

La idea de este árbol es construirlo recursivamente hacia arriba (desde abajo) con el criterio que se plantea en la cápsula para luego en la búsqueda, considerando el rango de tu criterio  simplemente buscar cómo un árbol binario (desde arriba), de esta manera cómo las hojas estarán ordenadas tendrás todos los valores en ese rango.

Espero se haya entendido.

**Reply by Juanfischermontero** [reply]
Hola, eso lo entiendo. Mi pregunta es que en el ejemplo no se esta siguiendo la busqueda de arbol binario o si? Se esta buscando el 3. El nodo raiz es 2. Con el criterio de busqueda binaria nos deberiamos ir hacia la derecha porque 2 < 3. Si embargo, como se ve en el dibujo se va a la izquierda, nuevamente se compara con un 2, ahora si se va a la derecha y encuentra el 3. Esta raro o hay algo que no estoy viendo?

**Reply by MatiLaB** [reply]
Creo que en la capsula se menciona que los padres de los nodos tendrán el mayor valor de la rama izquierda, y eso siguiendo para arriba, así que creo que la forma en que se hace la búsqueda está bien, solo que la raíz debería ser un 3 no un 2.
Eso es lo que yo entendí ahora revisando la capsula, pero ojalá que un ayudante confirme :]

**Reply by CarlangaUC** [reply]
Hola!

Efectivamente hay un typo ahi, nuevamente el concepto y la explicación asociada es lo importante, la idea es ocupar el **max. sub hijo izquierdo** de esta manera al ir construyendo recursivamente hacia arriba, tomaras el máximo hijo posible de cada subarbol, es por esto que como bien dicen debería ser **3** el nodo raíz ya que el subárbol izquierdo presenta como máximo hijo el 3.

Coménteme si les quedo mas claro 👍
